src/backend/marsha/core/views.py

```diff
@@ -588,6 +543,101 @@ def post(self, request, *args, **kwargs):
         return self.render_to_response(self.get_context_data())
 
 
+@method_decorator(csrf_exempt, name="dispatch")
+@method_decorator(xframe_options_exempt, name="dispatch")
+class LTIRespondView(TemplateResponseMixin, View):
+    """LTI view called to respond to a consumer.
+
+    ie. after a deep linking content selection.
+
+    """
+
+    template_name = "core/form_autosubmit.html"
+
+    # pylint: disable=unused-argument
+    def post(self, request, *args, **kwargs):
+        """Respond to POST request.
+
+        Renders a form autosubmitted to a LTI consumer with signed parameters.
+
+        Parameters
+        ----------
+        request : Request
+            passed by Django
+        args : list
+            positional extra arguments
+        kwargs : dictionary
+            keyword extra arguments
+        Returns
+        -------
+        HTML
+            generated from applying the data to the template
+
+        """
+        try:
+            jwt_token = AccessToken(self.request.POST.get("jwt", ""))
+            jwt_token.verify()
+
+            lti_select_form_data = jwt_token.payload.get("lti_select_form_data")
+            content_item_return_url = lti_select_form_data.get(
+                "content_item_return_url"
+            )
+
+            urlparse(content_item_return_url)
+
+            # filters out oauth parameters
+            lti_parameters = {
+                key: value
+                for (key, value) in lti_select_form_data.items()
+                if "oauth" not in key
+            }
+            lti_parameters.update(
+                {"content_items": self.request.POST.get("content_items")}
+            )
+
+            # generate signature
+            passport = LTIPassport.objects.get(
+                oauth_consumer_key=lti_select_form_data.get("oauth_consumer_key"),
+                is_enabled=True,
+            )
+            client = oauth1.Client(
+                client_key=passport.oauth_consumer_key,
+                client_secret=passport.shared_secret,
+            )
+            # Compute Authorization header which looks like:
+            # Authorization: OAuth oauth_nonce="80966668944732164491378916897",
+            # oauth_timestamp="1378916897",
+            # oauth_version="1.0", oauth_signature_method="HMAC-SHA1",
+            # oauth_consumer_key="",
+            # oauth_signature="frVp4JuvT1mVXlxktiAUjQ7%2F1cw%3D"
+            _uri, headers, _body = client.sign(
+                content_item_return_url,
+                http_method="POST",
+                body=lti_parameters,
+                headers={"Content-Type": "application/x-www-form-urlencoded"},
+            )
+
+        # if anything goes wrong, abort operation with PermissionDenied
+        except Exception as error:
```

Comment:
Your except handles to much cases I think. You should had one for each case you want to manage. Here I see `jwt.verify` and `LTIPassport.objects.get`. For each use the exact exception you expect. Everything else should lead to a 500 and an alert in sentry.